backends/backend-sdl/src/arc/backend/sdl/SdlApplication.java

```diff
@@ -260,8 +265,33 @@ public void exit(){
     }
 
     public static class SdlError extends RuntimeException{
-        public SdlError() {
+        public SdlError(){
             super(SDL_GetError());
         }
     }
+
+    /** MacOS doesn't work when -XstartOnFirstThread is not passed, this will restart the program with that argument if it isn't already present. */
+    private void restartMac(){
+        String id = ManagementFactory.getRuntimeMXBean().getName().split("@")[0];
+
+        if(!OS.hasEnv("JAVA_STARTED_ON_FIRST_THREAD_" + id) || OS.env("JAVA_STARTED_ON_FIRST_THREAD_" + id).equals("0")){ //check if equal to 0 just in case
+            Log.warn("Applying -XstartOnFirstThread for macOS support.");
+            String javaPath = //attempt to locate java
+                new Fi(OS.prop("java.home")).child("bin/java").exists() ? new Fi(OS.prop("java.home")).child("bin/java").absolutePath() :
+                Core.files.local("jre/bin/java").exists() ? Core.files.local("jre/bin/java").absolutePath() :
+                "java";
+            try{
+                Fi jar = Fi.get(SdlApplication.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath());
+                Seq<String> launchOptions = Seq.with(javaPath);
+                launchOptions.addAll(ManagementFactory.getRuntimeMXBean().getInputArguments());
+                launchOptions.addAll(System.getProperties().entrySet().stream().map(it -> "-D" + it).toArray(String[]::new));
+                launchOptions.addAll("-XstartOnFirstThread", "-jar", jar.absolutePath(), "-firstThread");
+
+                new ProcessBuilder(launchOptions.toArray(String.class)).inheritIO().start();
```

Comment:
A post I found on this suggest using `System.exit(process.waitFor());` so it returns the same exist code as the "real" process.

Reply:
I had something like that originally but then i get a random java window thats always open

Reply:
I tested this out some more with the current setup and it appears that it is no longer an issue.